gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java

```diff
@@ -278,7 +298,7 @@ public void handle(AsyncResult<HttpClientRequest> event) {
                 }
                 setProfileHeaders(log, profileHeaderMap, cReq);
 
-                authHeader.ifPresent(authHeaderValue -> cReq.headers().set(authHeaderValue.key(), authHeaderValue.value()));
+                if (authHeader != null) cReq.headers().set(authHeader.key(), authHeader.value());
```

Comment:
Why going from _Optional<AuthHeader>_ to _AuthHeader_ where you have to use _null_ checks again. I don't see how this makes the code better or more readable

Reply:
My opinion clearly differs in this regard. I just changed it back.

(HINT: Feel free to ignore my opinion that follows here)
<opinion>

Optional is nothing but adding unnecessary bloat in this case (*Just comparing both variants side-by-side should be enough to show why*). [I'm not alone](https://homes.cs.washington.edu/~mernst/advice/nothing-is-better-than-optional.html) with this opinion BTW. Yet another counter argument comes from IDEA itslef:

![image](https://github.com/swisspost/gateleen/assets/23085769/9477a04f-c0f6-4da7-91bb-98ad5cf72b49)

</opinion>